content/blog/trustless-lightning-payments-for-content.md

```diff
@@ -0,0 +1,357 @@
+Lightning network to reward content creators:
+
+- Value-For-Value (Used in Podcasting 2.0 apps like Fountain)
+- Tipping (Used in Zion)
+
+These apps use a voluntary payment model, where any user can access content
+for free, and then decide how much they want to pay in tips to reward the
+content creator.
+
+This leads to a situation where the majority of users are consuming the
+content for free, and a small generous minority of users is generating all
+of the revenue for the content creator. The revenue of the content creator
+is totally dependent on the generosity and altruism of their consumers.
+
+It is the digital equivalent of "busking" on a sidewalk, and hoping that people
+will drop a few coins in your hat.
+
+
+#### Moving to a pay for content model
+
+A better, fairer model would be one where the consumer is required to make a
+payment to get access to the content.
+
+A naive approach to accomplish this would be to simply encrypt the content,
+and then create a Lightning invoice with the decryption key as the invoice
+preimage (using HTLCs), and share the invoice with the consumer. Then, when the consumer
+pays the invoice, they would obtain the decryption key, and use it to unlock
+the content.
+
+<!-- In the current implementation of Lightning in LND (using HTLCs), the preimage -->
+<!-- would be specified in the `r_preimage` field of the `AddInvoice` RPC method. -->
+
+{{< center-figure src="/post-data/trustless-lightning-payments/htlc-example-3.png" caption="" height=600 width=600 >}}
+
+Unfortunately, this approach requires the consumer to trust the honesty of the
+content seller. A malicious seller could create an invoice using an invalid
+decryption key as the preimage. The buyer would make the Lightning payment, and
+only realize after the payment that they got an invalid decryption key. The
+buyer has no way to verify the validity of the invoice until it is too late,
+and they have already lost their funds.
+
+This model of monetization can only work if the buyer is able to trust the seller.
+This might be true if the seller is a large, reputable institution, but this
+model breaks down if we try to extend it to a peer-to-peer decentralized
+network, where all connections are assumed to be trustless.
+
+#### Making payments trustless with PTLCs
+
+Fortunately, there is a way to make the payment trustless for both the buyer
+and the seller, but it requires that Lightning use PTLC payment channels.
+
+With PTLCs, the seller specifies a 32-byte scalar value (`s`) as the "preimage"
+at the creation-time of the invoice. When the buyer decodes the invoice payment
+string, they will obtain the payment point (`p = s*G`). The buyer will only
+obtain the "preimage" `s` after the payment is settled.
+
+So now that we know that PTLC invoices use elliptic curve points, we can
+take advantage of the distributive property of points on elliptic curves:
+
+```
+(s1 + s2)*G = s1*G + s2*G
```

Comment:
I think you should use the LaTeX rendering for this and the other appearances of `s`, `p1`, etc above and below.

See the example blog post or the Schnorr Basic post for example usage. You'd have to enable LaTeX support in the front-matter with `mathjax: true`.

Reply:
Done.